### DISCUSSION TITLE
DATA ANALYSI, GRAPHES, S-R Trend Phases

### DISCUSSION THREAD

**Discussion by hfatima1** [lisphilar/covid19-sir#1002] [Q&A]
Is it possible to retrieve and download the data file that you uploaded?

I have lost the excel file that I uploaded for data analysis. Is there a way out that we can download our data file being uploaded?

Asking it because, I need to download the graphs and data into images form. (any code available for that?)

Also, with the file, I previously updated, it gave me S-R phase divided data into 31 phases and now it's giving me data into 9 phases, with these 9 phases, results are altered and simulations and scenario analysis curves are on steaper side.

**Comment by lisphilar** [comment]
CovsirPhy downloads [the recommended datasets](https://lisphilar.github.io/covid19-sir/markdown/LOADING.html#download-the-recommended-datasets) from some the external data sources. Because they are version-controlled at their GitHub repositories, we can get the old files, but this it is not a smart way.

`covsirphy.Scenario` class has `.timepoints(first_date=None, last_date=None, today=None)` method as documented in the following page.
https://lisphilar.github.io/covid19-sir/usage_phases.html#Preperation

As an example, the next script analyses only 01Jan2021-31Dec2021 even when the codes were performed on 17Feb2022.

```Python
import covsirphy as cs
loader = cs.DataLoader()
jhu_data = loader.jhu()
snl = cs.Scenario(country="Japan", province=None)
snl.register(jhu_data)
snl.timepoints(first_date="01Jan2021", today="31Dec2021")
snl.trend()
snl.summary()
# -> the start date of the 1st phase is 01Jan2021, the end date of the last phase is 31Dec2021
```

Note that the last date of `snl.records()` is still 17Feb2022 because `last_date` argument was not used. When specified, it will be changed.
